specs/Slots.md

```diff
@@ -0,0 +1,156 @@
+# Slots API specification
+
+## Slots definition
+
+Fluent UI components almost always contain sub parts and these pre-defined layouts can be configured via Slots API. Slots are named areas in a component that can receive props and provide styling and layout for them.
+
+For example, `Button` component contains `content`, `loader` and `icon` slots in its layout:
+
+```jsx
+// ⚠️ not a real JSX/DOM markup
+<slots.root>
+  <slots.loader />
+  <slots.children />
+  <slots.icon />
+</slots.root>
+```
+
+Slots let define the complex layouts required by many components according to accessibility and design requirements automatically. Slots can be configured through props, which lets the caller pass in a variety of inputs for a given slot.
+
+## Slots components
+
+Each slot is represented by a React component or a primitive, for example: `slots.loader` can be a `Loader` component and `slots.icon` can be a `span`:
+
+```jsx
+// ⚠️ not a real JSX/DOM markup
+<>
+  {/* renders a <button /> element */}
+  <slots.root>
+    {/* renders a <Loader /> element */}
+    <slots.loader />
+    {/* renders a <span /> element */}
+    <slots.icon />
+  </slots.root>
+</>
+```
+
+A default set of components is manually configured per each component, but is optional, by default slots are represented by `div` elements.
```

Comment:
Not quite sure what you're trying to say here, do you mean that is is possible to manually configure each slot ?

Reply:
Yes, may be you can suggest a better phrase there:
- components have slots
- each slot by default is `div` element
- this can be overridden

Reply:
```suggestion
Slots are rendered as `div` elements by default. However, all rendering aspects of a slot can be manually configured and overridden in the ways described below
```

Reply:
```suggestion
Slots are rendered as `div` elements by default. However, all rendering aspects of a slot can be manually configured and overridden.
```

I cut a part of phrase as it needs decisions from an upcoming RFC.